<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>618</width>
    <height>264</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label_26">
     <property name="text">
      <string>Type</string>
     </property>
     <property name="buddy">
      <cstring>typeCmb</cstring>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="4">
    <widget class="QComboBox" name="typeCmb">
     <property name="editable">
      <bool>false</bool>
     </property>
     <property name="currentText">
      <string>LoRA</string>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="maxVisibleItems">
      <number>2</number>
     </property>
     <property name="maxCount">
      <number>2</number>
     </property>
     <item>
      <property name="text">
       <string>LoRA</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>LoHA</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_27">
     <property name="text">
      <string>Base Model</string>
     </property>
     <property name="buddy">
      <cstring>baseModelLed</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="3">
    <widget class="QLineEdit" name="baseModelLed"/>
   </item>
   <item row="1" column="4">
    <widget class="QToolButton" name="baseModelBtn">
     <property name="text">
      <string>...</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_28">
     <property name="text">
      <string>Rank</string>
     </property>
     <property name="buddy">
      <cstring>rankSbx</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QSpinBox" name="rankSbx">
     <property name="maximum">
      <number>512</number>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <spacer name="horizontalSpacer_6">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>27</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="3" rowspan="3">
    <widget class="QFrame" name="doraFrm">
     <property name="frameShape">
      <enum>QFrame::Shape::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Plain</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QCheckBox" name="decomposeCbx">
        <property name="text">
         <string>Decompose Weights (DoRA)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="normCbx">
        <property name="text">
         <string>Use Norm Epsilon (DoRA Only)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="outputAxisCbx">
        <property name="text">
         <string>Apply On Output Axis (DoRA Only)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_29">
     <property name="text">
      <string>Alpha</string>
     </property>
     <property name="buddy">
      <cstring>alphaSbx</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QDoubleSpinBox" name="alphaSbx"/>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_30">
     <property name="text">
      <string>Dropout Probability</string>
     </property>
     <property name="buddy">
      <cstring>DropoutSbx</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_31">
     <property name="text">
      <string>Weight Data Type</string>
     </property>
     <property name="buddy">
      <cstring>weightDTypeCmb</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QComboBox" name="weightDTypeCmb"/>
   </item>
   <item row="6" column="0">
    <widget class="QCheckBox" name="bundleCbx">
     <property name="text">
      <string>Bundle Embeddings</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <spacer name="verticalSpacer_6">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>13</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="1">
    <widget class="QDoubleSpinBox" name="DropoutSbx">
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>typeCmb</tabstop>
  <tabstop>baseModelLed</tabstop>
  <tabstop>baseModelBtn</tabstop>
  <tabstop>rankSbx</tabstop>
  <tabstop>alphaSbx</tabstop>
  <tabstop>DropoutSbx</tabstop>
  <tabstop>weightDTypeCmb</tabstop>
  <tabstop>bundleCbx</tabstop>
  <tabstop>decomposeCbx</tabstop>
  <tabstop>normCbx</tabstop>
  <tabstop>outputAxisCbx</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>decomposeCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>normCbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>411</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>decomposeCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>outputAxisCbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>531</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
